rfcs/proposed/dynamic_selection_customization/README.md

```diff
@@ -0,0 +1,436 @@
+for SYCL queues is a bit more than 250 lines of code. With sensible defaults, this proposal aims to simplify backend writing to
+open up Dynamic Selection to more use cases.
+
+## An overview of the proposal
+
+This document proposes that there is a default backend that can be used with any resource type to get
+sensible default functionaility. In addition, custom backends can be created when needed by mixing in
+default functionality with specialized functionality. 
+
+### Proposed Defaults
+
+In the following subsections, we describe each
+of the functions and traits described in the [Current backend design](#current-backend-design) and
+describe the proposed default implementation and the implications of the defaults.
+
+#### Default `resource_t<T>` 
+
+Policies contain a backend and current policies are either default constructed/initialized or
+constructed/initialized with a `std::vector<Resouce>` of resources. When a vector is passed
+to the constructor, the type of the resource can be deduced and used as a template argument
+when constructing the backend. When a policy is default constructed, the resource type can be made
+a manditory template argument and this can be used to set the resource type in the backend.
+
+Is it therefore becomes unnecessary to explicitly provide the resource type in the default backend.
+For example, in the code below a `round_robin_policy` is constructed by passing a vector of
+pointers to `tbb::task_group`.  The type of the resource `tbb::task_group *` can be deduce and
+passed as a template argument to the default backend. 
+
+```cpp
+    tbb::task_group t1, t2;
+    ex::round_robin_policy p{ { &t1, &t2 } };
+```
+
+If the policy is default constructed or constructed for deferred initialization, the
+resource type must be given explicitly.
+
+```cpp
+    ex::round_robin_policy<tbb::task_group*> p1{ };
+    ex::round_robin_policy<tbb::task_group*> p2{ deferred_initialization_t };
+```
+#### Default `wait_t<T>`
+
+The `wait_t<T>` can only be deduced at the time a user function is passed to submit since
+it is the type returned by the user's function. There is no way to generally determine a
+`wait_t<T>` from an arbitrary `resource_t<T>` and therefore the default backend cannot
+provide a meaningful `wait_t<T>`.
+
+#### Default `b.submit(s, f, args…)`
+
+The main role that `submit` plays in a backend is to add instrumentation around the call to `f`.
+Experience has shown us that most backends perform four basic steps in their implementation
+of the `submit` function:
+
+1. Do any setup needed for implementing reporting before calling `f`.
+2. The function `f` is called and it return value captured.
+3. Do any setup needed for implementing reporting after calling `f`, perhaps using what was returned by `f`.
+4. The *Submission* object, which typically wraps what is returned by `f`, is constructed and returned.
```

Comment:
nitpick: 
This seems to go back and forth between present tense and past tense, we should keep the same orientation for all steps